### PR TITLE
Add tag option for package controller dev promote pipeline

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

*Issue #, if available:*

*Description of changes:*

Why this needs to exist is because there is two sets of dev build being produced, one by the EKS-A pipeline using Git Tag, and another by the new dev pipeline. The current logic just defaults to latest, so adding a tag option allows us to always promote our dev builds for e2e purposes to always use dev chart.

### CODE REVIEW

File: generatebundlefile/main.go [hunk 1]
```diff
@@ -91,9 +91,13 @@ func cmdPromote(opts *Options) error {
 
 	promoteCharts := make(map[string][]string)
```
Comment (TerryHowe) [inline generatebundlefile/main.go:93]: this doesn't make much sense

Comment (jonahjon) [reply generatebundlefile/main.go:93]: What doesn't make sense the `promoteCharts`?

Comment (jonahjon) [reply generatebundlefile/main.go:93]: It was added for scenario's where we are promoting multiple charts for a single pipeline like cluster-autoscaler, metrics server, emissary, metallb and we are using an input file to specify which ones to promote.

Comment (TerryHowe) [reply generatebundlefile/main.go:93]: well, if you pass in a tag, it gets set to latest. This is either wrong or the tag should be a boolean type

Comment (jonahjon) [reply generatebundlefile/main.go:93]: OOoooooooo, yea I did the logic backwards let me fix that.

Comment (jonahjon) [reply generatebundlefile/main.go:93]: Gonna open a new PR with the fix, and the script changes actually.

File: generatebundlefile/ecr_helper.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func ImageTagFilter(details []ImageDetailsBothECR, version string) []ImageDetail
 	var filteredDetails []ImageDetailsBothECR
 	for _, detail := range details {
 		for _, tag := range detail.ImageTags {
```
Comment (TerryHowe) [inline generatebundlefile/ecr_helper.go:108]: Our bundles are like `v1-24-latest` is that going to be a problem with this?

Comment (jonahjon) [reply generatebundlefile/ecr_helper.go:108]: No it wouldn't be a problem with that, this is only for promoting helm chart, so it's not using the k8s version in this logic at all.

I have it working locally, just working on adding some tests right now.

**Comment by jonahjon** [comment]
/hold
working on tests

File: generatebundlefile/main.go (hunk 1 above)
Comment (jonahjon) [reply generatebundlefile/main.go:93]: Fixed

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**